File: csharp/hexadecimal/Hexadecimal.cs
using System;

class Hexadecimal
{
  public static int ToDecimal(string hex)
  {
    var result = 0.0;
    for (var i = 0; i < hex.Length; i++)
    {
      if (!char.IsLetterOrDigit(hex[i]))
        return 0;
      if (char.IsDigit(hex[i]))
      {
        result += char.GetNumericValue(hex[i]) * Math.Pow(16, hex.Length - (i + 1));
      }
      else if (char.IsLetter(hex[i]))
      {
        var offset = char.ToLower(hex[i]) - 97;
        if (offset > 6)
          return 0;
        result += (10 + offset) * Math.Pow(16, hex.Length - (i + 1));
      }
    }
    return (int)result;
  }
}

File: javascript/hexadecimal/hexadecimal.js
var Hexadecimal = function (hexadecimal) {
  this.hexadecimal = hexadecimal.toLowerCase();
};

Hexadecimal.prototype.toDecimal = function () {
  var result = 0;
  for (var i = 0; i < this.hexadecimal.length; i++) {
    if ((this.hexadecimal[i] >= '0' && this.hexadecimal[i] <= '9') || (this.hexadecimal[i] >= 'a' && this.hexadecimal[i] <= 'f')) {
      var hexValue = (this.hexadecimal[i] >= 'a' && this.hexadecimal[i] <= 'f') ? this.hexadecimal.charCodeAt(i) - 87 : +this.hexadecimal[i];
      result += hexValue * Math.pow(16, this.hexadecimal.length - (i + 1));
    } else {
      return 0;
    }
  }
  return result;
};

module.exports = Hexadecimal;